core/src/main/java/org/elasticsearch/search/rescore/RescoreBuilder.java

```diff
@@ -112,27 +199,85 @@ public QueryRescorer setRescoreQueryWeight(float rescoreQueryWeight) {
         }
 
         /**
-         * Sets the original query score mode. The default is <tt>total</tt>
+         * Gets the original query weight for rescoring. The default is <tt>1.0</tt>
+         */
+        public float getRescoreQueryWeight() {
+            return this.rescoreQueryWeight;
+        }
+
+        /**
+         * Sets the original query score mode. The default is {@link QueryRescoreMode#Total}.
+         * @deprecated use {@link RescoreBuilder.QueryRescorer#setScoreMode(QueryRescoreMode)}
          */
+        @Deprecated
         public QueryRescorer setScoreMode(String scoreMode) {
+            return setScoreMode(QueryRescoreMode.fromString(scoreMode));
+        }
+
+        /**
+         * Sets the original query score mode. The default is {@link QueryRescoreMode#Total}.
+         */
+        public QueryRescorer setScoreMode(QueryRescoreMode scoreMode) {
             this.scoreMode = scoreMode;
             return this;
         }
 
+        /**
+         * Gets the original query score mode. The default is <tt>total</tt>
+         */
+        public QueryRescoreMode getScoreMode() {
+            return this.scoreMode;
+        }
+
         @Override
         protected XContentBuilder innerToXContent(XContentBuilder builder, Params params) throws IOException {
             builder.field("rescore_query", queryBuilder);
-            if (queryWeight != null) {
-                builder.field("query_weight", queryWeight);
-            }
-            if (rescoreQueryWeight != null) {
-                builder.field("rescore_query_weight", rescoreQueryWeight);
+            builder.field("query_weight", queryWeight);
+            builder.field("rescore_query_weight", rescoreQueryWeight);
+            builder.field("score_mode", scoreMode.name().toLowerCase(Locale.ROOT));
```

Comment:
I don't see any parsing code changes in this PR so maybe thats coming in a later one, but when you do the parsing code changes can you convert these to use ParseField.getPreferredName() ?